common/rfb/Exception.h

```diff
@@ -18,17 +18,22 @@
 #ifndef __RFB_EXCEPTION_H__
 #define __RFB_EXCEPTION_H__
 
-#include <rdr/Exception.h>
+#include <stdexcept>
 
 namespace rfb {
-  typedef rdr::Exception Exception;
-  struct AuthFailureException : public Exception {
+  class protocol_error : public std::runtime_error {
+  public:
+    protocol_error(const char* what_arg) : std::runtime_error(what_arg) {}
+    protocol_error(const std::string& what_arg) : std::runtime_error(what_arg) {}
+  };
+
+  struct AuthFailureException : public std::runtime_error {
```

Comment:
Did the `AuthFailureException` and `AuthCancelledException` end up in the wrong commit here?

Reply:
I split things a bit more now to make it more clear. Please have a look.